data/src/main/java/com/microsoft/azure/kusto/data/auth/endpoints/KustoTrustedEndpoints.java

```diff
@@ -0,0 +1,144 @@
+package com.microsoft.azure.kusto.data.auth.endpoints;
+
+import com.microsoft.azure.kusto.data.Ensure;
+import com.microsoft.azure.kusto.data.auth.CloudInfo;
+import com.microsoft.azure.kusto.data.exceptions.DataServiceException;
+import com.microsoft.azure.kusto.data.exceptions.KustoClientInvalidConnectionStringException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.*;
+import java.util.function.Predicate;
+
+/**
+ * A helper class to determine which DNS names are "well-known/trusted"'
+ * Kusto endpoints. Untrusted endpoints might require additional configuration
+ * before they can be used, for security reasons.
+ */
+public class KustoTrustedEndpoints {
+    private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    public static boolean enableWellKnownKustoEndpointsValidation = true;
+    private static final Map<String, FastSuffixMatcher> matchers;
+    private static FastSuffixMatcher additionalMatcher;
+    private static Predicate<String> overrideMatcher; // We could unify this with matchers, but separating makes debugging easier
+
+    static {
+        matchers = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+
+        WellKnownKustoEndpointsData.getInstance().AllowedEndpointsByLogin.forEach((key, value) -> {
```

Comment:
Here you can also use streams, but it's up to you if it's more readable:
```java
        matchers = WellKnownKustoEndpointsData.getInstance().AllowedEndpointsByLogin.entrySet().stream()
                .collect(Collectors.toMap(Map.Entry::getKey, e -> FastSuffixMatcher.Create(
                        Stream.concat(
                                e.getValue().AllowedKustoHostnames.stream().map(hostname -> new MatchRule(hostname, true)),
                                e.getValue().AllowedKustoSuffixes.stream().map(suffix -> new MatchRule(suffix, false)))
                                .collect(Collectors.toList())
                )));
```

Reply:
i don't like using it just for the sake of using it - this example is too much for me